control_client/src/pad.c

```diff
@@ -67,11 +74,24 @@ int pad_connect_forever(pad_t *pad) {
 /*
  * Send a message to the control port of the pad server.
  * @param pad The pad server to send a message to.
- * @param buf The buffer containing `n` bytes of message to be sent.
- * @param n The number of bytes in `buf` to be sent.
+ * @param iov 2 iovec structs, one for the header and one for the payload.
  * @return The number of bytes that were sent, or -1 on failure (errno indicates the error).
  */
-ssize_t pad_send(pad_t *pad, const void *buf, size_t n) { return send(pad->sock, buf, n, 0); }
+ssize_t pad_send(pad_t *pad, struct iovec iov[2]) {
```

Comment:
I think it would be better to support a `struct iovec *iov` and an `iovlen` in case we ever need more flexibility, but this is not a requirement for merge, we can add that when we need it.

Reply:
Very fair, I'm was also thinking about reverting some changes to that this PR only keeps the pad_server acknowledgement logic